pandas-stubs/core/tools/datetimes.pyi

```diff
@@ -38,7 +38,7 @@ DatetimeScalar: TypeAlias = Scalar | datetime | np.datetime64 | date
 
 DatetimeScalarOrArrayConvertible: TypeAlias = DatetimeScalar | ArrayConvertible
 
-DatetimeDictArg: TypeAlias = list[Scalar] | tuple[Scalar, ...] | AnyArrayLike
+DatetimeDictArg: TypeAlias = list[Scalar] | tuple[Scalar, ...] | AnyArrayLike | int
```

Comment:
There are 2 ways of doing this:

1. Use `Scalar` instead of `int` so it mirrors the rest of the declaration.
2. Change `list[Scalar]`, `tuple[Scalar, ...]` and `AnyArrayLike` to be more restrictive.  For example, you can't pass `float` values, and maybe just `int` and `str` are accepted.  So right now the type is too wide.

I'm OK if you just want to do (1) for now.

Reply:
So it turns out you can actually pass some float but it is a weird edge case and I would advise against it:
```python
time_series = pd.date_range("2023-10-01", "2024-05-01", freq="1h")
pd.to_datetime({"year": time_series.year, "month": 3, "day": 4.5})
```
To me this 4.5 at the end just gets converted to 4 so I would advise against allowing float, also no other tests fail with switching to `int | str` instead of `float | str`, please let me know if you agree with that.

Reply:
Wrt `AnyArrayLike`, it can't be subscribed as a generic but indeed ideally it should be something like `AnyArrayLike[int | str]`, prob an issue for another day.